### PR TITLE
4.11 Support

### PR DESCRIPTION
**Pull request by SudoBrendan**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/14592825 - pulled from https://github.com/Azure/ARO-RP/pull/2339

### What this PR does / why we need it:

Adds support for OpenShift 4.11 in multi-version.

- [x] `hack/update-go-mod-dependencies.go`, `go.mod`, `go.sum`, `/vendor`, and `pkg/util/version/const.go`: Support OpenShift 4.11
    - [x] Bump the [installer](https://github.com/jewzaam/installer-aro/tree/release-4.11-azure) to 4.11
    - [x] Fix module dependency issues between the installer, its dependencies, and our RP's dependencies
    - [x] Use golang 1.18
    - [x] Bump the default RHCOS images when installing new clusters
        - [x] [Publish the images](https://github.com/Azure/ARO-RP/blob/master/docs/publish-rhcos-image.md)
    - [x] Bump the container image tag for OCP manifests
- [x] `/pkg/installer`, `/pkg/cluster`: Update RP codebase to support new infrastructure changes and installer changes
    - [x] `/pkg/installer/generateconfig.go`
        - [x] native OpenShift installer now supports DiskEncryptionSets, so we've removed our patch from the installer and bumped the RP code to use their implementation instead of ours
        - [x] disable [baremetal api capability](https://pkg.go.dev/github.com/openshift/api/config/v1#ClusterVersionCapabilitySet), which is now a default Capability
        - [ ] ~~bootstrapper VM is now sized the same as masters, and supports encryption~~ (subsequent PR)
        - [ ] ~~master VMs now support Ultra SSD Data Disks~~ (subsequent PR)
        - [ ] ~~master VMs now support accelerated networking~~ (subsequent PR - forcing `Basic` networking for now on all VM types - default is to use Accel if supported on the type)
        - [ ] ~~adding support for HyperV Gen 2~~ (subsequent PR)
        - [ ] ~~Set OVN as the default CNI~~ (subsequent PR)
    - [ ] `/pkg/cluster/storageclass.go`: `managed-premium` StorageClass has been removed. If Encryption is set, create a `managed-csi` encrypted equivalent and make it the default. **This file doesn't exist in this repo**, so it looks like we don't do anything special with StorageClasses in this kind of install - what are the implications of this for cluster install features and our E2E suite?

### Test plan for issue:

> NOTE: :warning: I'm not sure how to test these changes, this PR is untested by Brendan.

- [ ] Cluster Installer
    - [ ] test default create
    - [ ] test with encryption

### CODE REVIEW

**Comment by cblecker** [comment]
/hold
for review

File: pkg/installer/generateconfig.go
Comment (darthhexx) [inline pkg/installer/generateconfig.go:124]: The call to ParseResourceID fails in the default casze when user encryption keys are not provided. I reworked this in the version that I tested successfully on 4.10 and 4.11 with the following changes:

```suggestion
	var masterDiskEncryptionSet *azuretypes.DiskEncryptionSet
	var workerDiskEncryptionSet *azuretypes.DiskEncryptionSet

	if m.oc.Properties.MasterProfile.DiskEncryptionSetID != "" {
		masterDiskEncryptionSetResource, err := azure.ParseResourceID(m.oc.Properties.MasterProfile.DiskEncryptionSetID)
		if err != nil {
			return nil, nil, err
		}

		masterDiskEncryptionSet = &azuretypes.DiskEncryptionSet{
			SubscriptionID: masterDiskEncryptionSetResource.SubscriptionID,
			ResourceGroup:  masterDiskEncryptionSetResource.ResourceGroup,
			Name:           masterDiskEncryptionSetResource.ResourceName,
		}
	}

	if m.oc.Properties.WorkerProfiles[0].DiskEncryptionSetID != "" {
		workerDiskEncryptionSetResource, err := azure.ParseResourceID(m.oc.Properties.WorkerProfiles[0].DiskEncryptionSetID)
		if err != nil {
			return nil, nil, err
		}

		workerDiskEncryptionSet = &azuretypes.DiskEncryptionSet{
			SubscriptionID: workerDiskEncryptionSetResource.SubscriptionID,
			ResourceGroup:  workerDiskEncryptionSetResource.ResourceGroup,
			Name:           workerDiskEncryptionSetResource.ResourceName,
		}
	}
```

File: pkg/installer/generateconfig.go
Comment (darthhexx) [inline pkg/installer/generateconfig.go:167]:
```suggestion
							DiskEncryptionSet: masterDiskEncryptionSet,
```

File: pkg/installer/generateconfig.go
Comment (darthhexx) [inline pkg/installer/generateconfig.go:190]:
```suggestion
								DiskEncryptionSet: workerDiskEncryptionSet,
```

**Review by darthhexx — CHANGES_REQUESTED** [review]
With the suggested changes, and the [related PR for the RP](https://github.com/Azure/ARO-RP/pull/2585), I was able to successfully create Hive installed clusters for 4.10.20 and 4.11.16 with both default and user provided disk encryption keys.

File: pkg/util/version/const.go
Comment (darthhexx) [inline pkg/util/version/const.go:31]: I confirmed that this PR works together with the [related RP PR](https://github.com/Azure/ARO-RP/pull/2585) for 4.11.16. Would you mind bumping to that version?

**Review by darthhexx — CHANGES_REQUESTED** [review]

**Review by darthhexx — APPROVED** [review]
I have been able to successfully create 4.11 clusters using this PR and the [related ARO-RP PR](https://github.com/Azure/ARO-RP/pull/2585). Once both are merged, I would be comfortable to start enabling multi-version with 4.10 and 4.11.

File: pkg/installer/generateconfig.go
Comment (cblecker) [inline pkg/installer/generateconfig.go:161]: Is there a reason we are forcing this to basic?

Comment (SudoBrendan) [reply pkg/installer/generateconfig.go:161]: because there were concerns from team members about us implementing accelerated networking in the past that led to outages - it was suggested that we put any accelerated networking capabilities behind a toggle-able flag we could revert at any time, which would require an API release.

**Comment by cblecker** [comment]
/hold cancel

The other items will be done in follow ups.

**Pull request merged**
